### PR TITLE
mongocli 1.16.0 (new formula)

### PR DESCRIPTION
**Pull request by gssbzn**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/mongocli.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  url "https://github.com/mongodb/mongocli/archive/refs/tags/v1.16.0.tar.gz"
+  sha256 "184f0a13ca0cfb642203727b4971622a761c77ac85066fc51689356690f5e2d0"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    with_env(
+      MCLI_VERSION: version.to_s,
+      MCLI_GIT_SHA: "0f7699e45eca5e715b33822a8755d3b3b1469aaa",
```
Comment (carlocab) [inline Formula/mongocli.rb:13]: How is this updated?

File: Formula/mongocli.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    ) do
+      system "make", "build"
+    end
+    bin.install "bin/mongocli"
+
+    (bash_completion/"mongocli").write `#{bin}/mongocli completion bash`
```
Comment (carlocab) [inline Formula/mongocli.rb:19]:
```suggestion
    (bash_completion/"mongocli").write Utils.safe_popen_read(bin/"mongocli", "completion", "bash")
```
Same suggestion for the two lines below.

File: Formula/mongocli.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    (fish_completion/"mongocli.fish").write `#{bin}/mongocli completion fish`
+    (zsh_completion/"_mongocli").write `#{bin}/mongocli completion zsh`
+  end
+
+  test do
+    assert_match "mongocli version: #{version}", shell_output("#{bin}/mongocli --version")
```
Comment (carlocab) [inline Formula/mongocli.rb:25]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Comment by carlocab** [comment]
Thanks for the PR, @gssbzn. However, `mongocli` currently doesn't meet the notability requirement for inclusion into Homebrew/core:

```
mongocli:
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
```

I imagine it will at some point soon, but we should probably wait until that happens.

**Comment by nandahkrishna** [comment]
@gssbzn I think it will now pass the audit as there are 75 stars. You may have to open a new PR.

**Comment by gssbzn** [comment]
I'm going to try to open a new PR tomorrow that address some of the comments but I'm still thinking about what todo with the gitsha, that was one of the things I wanted to check here and why I open the PR before time to see what was the best approach for somethign like it but I'm thinking maybe I can add a dummy file on the packaging that can be read for it

**Comment by carlocab** [comment]
> I can add a dummy file on the packaging that can be read for it

This might work well, so that it doesn't need to be updated manually each time.

**Comment by gromgit** [comment]
AFAICT, your project doesn't actually use `MCLI_GIT_SHA` to influence the build process, so why not set it to a fixed string like `"release"`? I believe that's "standard procedure" for all current formulae that embed both version and commit SHA in their binaries, since the Git tag corresponding to the version already uniquely identifies the commit.

**Comment by gssbzn** [comment]
> so why not set it to a fixed string like "release"? 

also thought of it, if the brew team agrees I may go with that then

**Pull request closed without merging**